db-config.php

```diff
@@ -284,37 +284,48 @@ function my_db_callback($query, $wpdb) {
  * @param string $password Database password.
  */
 /*
-function add_db_server($dataset, $part, $dc, $read, $write, $host, $lhost, $name, $user, $password, $timeout = 0.2 ) {
+function add_db_server( $dataset, $part, $dc, $read, $write, $host, $lhost, $name, $user, $password, $timeout = 0.2 ) {
 	global $wpdb;
 
 	// dc is not used in hyperdb. This produces the desired effect of
 	// trying to connect to local servers before remote servers. Also
 	// increases time allowed for TCP responsiveness check.
-	if ( !empty($dc) && defined(DATACENTER) && $dc != DATACENTER ) {
-		if ( $read )
+	if ( ! empty( $dc ) && defined( 'DATACENTER' ) && $dc != DATACENTER ) {
+		if ( $read ) {
 			$read += 10000;
-		if ( $write ) 
+		}
+
+		if ( $write ) {
 			$write += 10000;
+		}
+
 		$timeout = 0.7;
 	}
 
 	// You'll need a hyperdb::add_callback() callback function to use partitioning.
 	// $wpdb->add_callback( 'my_func' );
-	if ( $part )
+	if ( $part ) {
 		$dataset = $dataset . '_' . $part;
+	}
 
-	$database = compact('dataset', 'read', 'write', 'host', 'name', 'user', 'password', 'timeout');
+	$database = compact( 'dataset', 'read', 'write', 'host', 'name', 'user', 'password', 'timeout' );
 
-	$wpdb->add_database($database);
+	$wpdb->add_database( $database );
 
 	// lhost is not used in hyperdb. This configures hyperdb with an
 	// additional server to represent the local hostname so it tries to
 	// connect over the private interface before the public one.
-	if ( !empty( $lhost ) ) {
-		if ( $read )
-			$database['read'] = $read - 0.5;
-		if ( $write )
-			$database['write'] = $write - 0.5;
+	if ( ! empty( $lhost ) && $lhost !== $host ) {
+		$database['host'] = $lhost;
```

Comment:
Similarly, this line being missing previously meant that `$lhost` didn't work in this example.